File: tps/tp1/ex16/PalindromoRecursivo.java
import java.util.Scanner;

public class PalindromoRecursivo {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String palavra = sc.nextLine();

        while (!palavra.equals("FIM")) {
            if (ehPalindromo(palavra, 0, palavra.length() - 1)) {
                System.out.println("SIM");
            } else {
                System.out.println("NAO");
            }
            palavra = sc.nextLine(); 
        }

        sc.close();
    }

    public static boolean ehPalindromo(String palavra, int esquerda, int direita) {
        if (esquerda >= direita) {
            return true;
        }
        if (palavra.charAt(esquerda) != palavra.charAt(direita)) {
            return false;
        }
        return ehPalindromo(palavra, esquerda + 1, direita - 1);
    }
}

File: tps/tp1/ex17/PalindromoRecursivo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool ehPalindromo(char palavra[256]);

int main()
{
    char palavra[256];
    fgets(palavra, sizeof(palavra), stdin);
    while( strcmp(palavra, "FIM\n") != 0)
    {
        if(ehPalindromo(palavra))
        {
            printf("SIM\n");
        }
        else{
            printf("NAO\n");
        }
        fgets(palavra, sizeof(palavra), stdin);
    }
}

bool ehPalindromo(char palavra[256])
{
    int tamanho = 0;
    while(palavra[tamanho] != '\n')
    {
        tamanho++;
    }
    for(int i = 0 ; i < tamanho; i++)
    {
        if(palavra[i] != palavra[tamanho -i -1])
        {
            return false;
        }
    }
    return true;
}